launch_ros/launch_ros/actions/lifecycle_node.py

```diff
@@ -42,8 +42,9 @@ class LifecycleNode(Node):
     def __init__(
         self,
         *,
-        name: Optional[Text] = None,
-        node_name: Optional[Text] = None,
+        name: Optional[SomeSubstitutionsType] = None,
+        namespace: SomeSubstitutionsType,
+        node_name: Optional[SomeSubstitutionsType] = None,
```

Comment:
@ivanpauno how's that `name` is optional but `namespace` isn't?

Reply:
https://github.com/ros2/launch_ros/blob/e8894ce6155b516b6c5f4dda1c5b4cba76c3e725/launch_ros/launch_ros/actions/lifecycle_node.py#L80-L89

Reply:
Hmm, right, the deprecated argument. It was [deprecated on Foxy](https://github.com/ros2/launch_ros/blob/3775aa3255ddb8c7271d47c71f08dd0f7cf14d2c/launch_ros/launch_ros/actions/node.py#L129-L131) though, so I think we can drop it (on a follow-up a PR).